src/classes/CW_AudienceCSVProcessHelper.cls

```diff
@@ -0,0 +1,406 @@
+        //for(CW_Reward_Library__c fbRec: [SELECT ID,FreeBet_Id__c FROM CW_Reward_Library__c WHERE FreeBet_Id__c=:FreeBetIdmap.KeySet() AND Reward_Catalogue__c= :CsvStatus.Reward_Catalogue__c ]){
+        //		FreeBetIdmap.Put(ToUpper(fbRec.FreeBet_Id__c),fbRec.ID);
+        //}
+        
+        for(CW_CSV_Stage_AudienceMembers__c rec:MembrRewardsMap.Values()){
+        		rec.Audience_Group_ref__c =GroupsIdmap.get(ToUpper(rec.Group_Name__c));
+        		rec.Contact_ref__c        =PlayerIdmap.get(ToUpper(rec.PlayerId__c));
+        		rec.Reward_Library__c     =FreeBetIdmap.get(ToUpper(rec.FreeBetId__c));
+        }
+        Database.insert(MembrRewardsMap.Values());
+    }
+    
+    
+   public void processCreateSheildAccounts( List<CW_CSV_Stage_AudienceMembers__c> scope){
+   	CsvStageList=scope;
+   	map<string,string> PlayrsMap= new map<string,string>();
+   	for(CW_CSV_Stage_AudienceMembers__c rec:CsvStageList ){
+   		PlayrsMap.Put(rec.PlayerId__c.ToUpperCase(),rec.PlayerId__c);
+   	}
+   	map<string,Id> PlayrsIdMap=CreateSheildAccounts(new Set<string>(PlayrsMap.values()));
+   	
+   	for(CW_CSV_Stage_AudienceMembers__c rec:CsvStageList ){
+   		rec.Contact_ref__c=PlayrsIdMap.get(rec.PlayerId__c.ToUpperCase());
+   	}
+   	update CsvStageList;
+   }
+   
+   //Step 2  - create Accounts and groups.
+   
+   public void processCreateAudienceGroups(List<CW_CSV_Stage_AudienceMembers__c> scope){
+   	CsvStageList=scope;
+   	map<string,string> groupMap= new map<string,string>();
+   	for(CW_CSV_Stage_AudienceMembers__c rec:CsvStageList ){
+   		groupMap.Put(rec.Group_Name__c.ToUpperCase(),rec.Group_Name__c);
+   	}
+   	map<string,Id> groupIdMap=CreateAudienceGroups(new Set<string>(groupMap.values()));
+   	
+   	for(CW_CSV_Stage_AudienceMembers__c rec:CsvStageList ){
+   		rec.Audience_Group_ref__c=groupIdMap.get(rec.Group_Name__c.ToUpperCase());
+   	}
+   	update CsvStageList;
+   }
+    
+    
+   public void processImportDataToCamp(List<CW_CSV_Stage_AudienceMembers__c> scope){
+   	//We have set groups , Members adn freeBets. So we can impotr all with seted reference others records a ignored.
+   	CsvStageList=scope;
+   	//firts step crate non exited camping rewards and load existings
+   	loadCreateCampaignReward();
+   	
+   	//Next step is  CreateAudiece sschedule
+   	createLoadAudienceGroupSchdule();
+   	
+   	createAudinceGroupScheduleReward();
+   	
+   	//And olmost FInala Step Is  
+   	createAudinceMember();
+   	//And Final step
+ 		createAudinceMemberReward();
+   } 
+    
+    
+   //Private methods for step 2 
+   private map<string,Id> CreateSheildAccounts(Set<String> ContactNameList){
+
+  // Rigth now crate Scheiled Account.
+    	List<Account> newContacs= new List<Account>();
+      if(ContactNameList.Size()>0 ){
+      	Id personAccountRTId = [select id from RecordType where SobjectType = 'Account' AND IsPersonType = true limit 1].id;
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.